NDTensors/src/default_storage.jl

```diff
@@ -0,0 +1,14 @@
+## This is a fil which specifies the default storage type provided some set of parameters
+## The parameters are the element type and storage type
+default_datatype(eltype::Type{<:Number}) = Vector{eltype}
+default_eltype() = Float64
+
+default_storagetype(datatype::Type{<:AbstractArray{ElT}}) where {ElT} = Dense{ElT,datatype}
+
+#currently similartype does not work for cuvector but the goal is for it to work with 
```

Comment:
Is it sufficient to add a definition:
```julia
buffertype(arraytype::Type{<:CuArray{<:Any,<:Any,B}}) where {B} = B
similartype(arraytype::Type{<:CuArray}, eltype::Type) = CuArray{eltype,ndims(arraytype),buffertype(arraytype)}
```
to handle that case? Same question to you @emstoudenmire since I know you hit a similar issue.

Also note that `CuArray` has an extra type parameter beyond `Array` which is a buffer type: https://github.com/JuliaGPU/CUDA.jl/blob/v3.13.1/src/array.jl#L25 which already speaks to why this in general needs to be overloaded for different `AbstractArray` types.